core/src/environments/godot_environment.py

```diff
@@ -1,65 +1,111 @@
+        high=np.array([2**60] * environment_settings.observation_space_size),
+        shape=(environment_settings.observation_space_size,),
+        dtype=np.float32,
     )
 
     def __init__(self, config: dict | None = None):  # noqa: ARG002
-        self._state: np.ndarray | None = None
+        super().__init__()
+        self._agent_ids = set(range(environment_settings.number_of_agents))
+        self._states: MultiAgentDict | None = None
         self.godot_handler = GodotHandler()
         self.godot_handler.launch_godot()
 
-    def step(self, action: np.integer) -> tuple[np.ndarray, float, bool, bool, dict]:
-        """
-        :param action: Action to be performed in the environment.
-        :return: tuple of:
-        State of the environment after performing the action,
-        Reward for performing the action,
-        Whether the game ended or not,
-        If truncated,
-        Extra info
+    def step(self, actions: MultiAgentDict):
+        """Returns observations from ready agents.
+
+        The returns are dicts mapping from agent_id strings to values. The
+        number of agents in the env can vary over time.
+
+        Returns:
+            Tuple containing 1) new observations for
+            each ready agent, 2) reward values for each ready agent. If
+            the episode is just started, the value will be None.
+            3) Terminated values for each ready agent. The special key
+            "__all__" (required) is used to indicate env termination.
+            4) Truncated values for each ready agent.
+            5) Info values for each agent id (maybe empty dicts).
         """
-        self.godot_handler.send(action.tobytes())
+        actions_serializable = [
+            {"id": key, "accelerate": value["accelerate"], "rotate": value["rotate"]} for key, value in actions.items()
+        ]
+
+        actions_json = json.dumps(actions_serializable)
+        self.godot_handler.send(actions_json.encode("utf-8"))
         return self.get_data()
 
     def get_data(self):
-        data: dict = self.godot_handler.request_data()
-        state = data["state"]
-        reward: float = data["reward"]
-        is_done: bool = data["is_done"]
-        truncated = False
-        info = {}
-        return state, reward, is_done, truncated, info
-
-    def reset(self, **_kwargs) -> tuple[np.ndarray, dict]:
-        """Resets the state of the environment and returns initial observations."""
-        return self.default_state, {}
+        try:
+            data_list = self.godot_handler.request_data()
+        except json.JSONDecodeError:
+            raise
+
+        states = {
+            data["Id"]: np.array(
+                [
+                    data["Speed"],
+                    data["Energy"],
+                    data["Health"],
+                    data["DistanceToClosestFood"],
+                    data["AngleToClosestFood"],
+                ]
+            )
```

Comment:
I think in the future the better way of doing this would be to send actions and receive observations from godot as vectors per agent instead of dictionaries. But for now, let's leave it as is.